sympy/polys/matrices/domainmatrix.py

```diff
@@ -444,8 +600,12 @@ def add(A, B):
             raise ShapeError("shape")
         if A.domain != B.domain:
             raise ValueError("domain")
+        if isinstance(A.rep, list) != isinstance(B.rep, list):
+            raise ValueError("format")
```

Comment:
This could get repetitive so maybe add a `_check` method like `DDM` has.

Reply:
Ok, I'll add it